packages/backend/src/graphql/resolvers/asset.ts

```diff
@@ -192,6 +197,58 @@ export const getFees: AssetResolvers<ApolloContext>['fees'] = async (
   }
 }
 
+export const deleteAsset: MutationResolvers<ApolloContext>['deleteAsset'] =
+  async (
+    _,
+    args,
+    ctx
+  ): Promise<ResolversTypes['DeleteAssetMutationResponse']> => {
+    try {
+      const assetService = await ctx.container.use('assetService')
+      const assetOrError = await assetService.delete({
+        id: args.input.id,
+        deletedAt: new Date().toISOString()
+      })
+
+      if (isAssetError(assetOrError)) {
+        switch (assetOrError) {
+          case AssetError.UnknownAsset:
+            return {
+              code: errorToCode[AssetError.UnknownAsset].toString(),
+              message: errorToMessage[AssetError.UnknownAsset],
+              success: false
+            }
+          case AssetError.InuseAsset:
+            return {
+              code: errorToCode[AssetError.InuseAsset].toString(),
+              message: errorToMessage[AssetError.InuseAsset],
+              success: false
+            }
+          default:
+            throw new Error(`AssetError: ${assetOrError}`)
+        }
```

Comment:
```suggestion
        return {
          code: errorToCode[assetOrError].toString(),
          message: errorToMessage[assetOrError],
          success: false
        }
```
I think you can get away with this

Reply:
not really :) 
that is an unreachable code section because of the switch: default, and assetOrError is an object so I cannot use it as an index

Reply:
Oh maybe the diff didn't show well, but I meant to replace the whole switch statement with that block. Typescript knows after `isAssetError` type guard that `assetOrError` is a string